features/finders.feature

```diff
@@ -24,12 +24,19 @@ Feature: Filtering documents
 
   Scenario: Filter document by keyword
     Given a collection of documents exist
-    When I search documents by keyword
+    When I search documents by keyword: "keyword searchable"
     Then I see all documents which contain the keywords
     And there is not a zero results message
     And the page title is updated
     And I can see that Google won't index the page
 
+  @javascript
+  Scenario: User tries to filter with malicious input
```

Comment:
Did you find a way to test this locally?

As in, make a change that means a raw `<script>` tag does appear on the page and see the final step fails? I had a go myself - just putting an `alert()` in the ERB file, except something seems to be scrubbing it. I can get it to work for the initial page load - requires a corresponding `accept_confirm` in the `visit` method call - but not for the actual search.

Looking at the page content, a significant amount of the structure is added or changed by JS, but I couldn't see anywhere that would be filtering out this tag.

Do you have any ideas?

Reply:
I was able to replicate it with a rather contrived bit of code. You can add `document.write(this.currentKeywords())`.

With the way this functionality has changed there is no longer the embedding of the search term via ruby in the live search (it's all done through JS dom methods currently) so it'd be quite hard to have an XSS issue. The controller test is the only one where ERB embedding is in play.

I could well be persuaded to remove this test, what do you think? I see it's inclusion somewhat justified by it being a regression test, however it is essentially a regression test for behaviour that is now removed.

Reply:
> I was able to replicate it with a rather contrived bit of code.

I'm not sure what you mean here. I was able to get an alert dialogue but only on the initial page load, not after the form submission. The removal of `<script>` tags - but not other tags - remains mysterious.

However, I'm not sure it matters as what you say makes sense: injection isn't a problem if it happens via JS, and if JS isn't enabled then any injection has no consequence.

While it's annoying there's no way to make the test fail today, I'm happy to think of it as an omission from before live search existed - it would likely still be here if it had been added originally.

And like you say, there's historical precedent for being overcautious here.